### PR TITLE
input_common: joycon: Remove magic numbers from calibration protocol

### PR DESCRIPTION
**Pull request by german77**
This PR doesn't introduce anything new. It's just a cleanup of the calibration protocol, it removes magic values and simplifies overall logic making it easier to read.

### CODE REVIEW

File: src/input_common/helpers/joycon_protocol/common_protocol.h [hunk 1]
```diff
@@ -100,7 +100,25 @@ class JoyconCommonProtocol {
      * @param size in bytes to be read
      * @returns output buffer containing the responce
      */
-    DriverResult ReadSPI(CalAddr addr, u8 size, std::vector<u8>& output);
+    DriverResult ReadSPI(SpiAddress addr, u8 size, std::vector<u8>& output);
+
+    template <typename Output>
+    DriverResult ReadSPI(SpiAddress addr, Output& output) {
```
Comment (liamwhite) [inline src/input_common/helpers/joycon_protocol/common_protocol.h:106]: `requires(std::is_trivially_copyable_v<Output>)`

File: src/input_common/helpers/joycon_protocol/joycon_types.h [hunk 2]
```diff
@@ -396,17 +395,49 @@ struct MotionData {
...
+struct SpiMagicCalibration {
+    CalibrationMagic first;
+    CalibrationMagic second;
+};
+static_assert(sizeof(SpiMagicCalibration) == 0x2, "SpiMagicCalibration is an invalid size");
+
+// Output from SPI read command containing joystick calibration
+struct JoystickSpiCalibration {
+    union {
+        struct left {
```
Comment (liamwhite) [inline src/input_common/helpers/joycon_protocol/joycon_types.h:408]:
```suggestion
        struct Left {
```

File: src/input_common/helpers/joycon_protocol/common_protocol.h [hunk 3]
```diff
@@ -100,7 +100,25 @@ class JoyconCommonProtocol {
...
+        std::vector<u8> buffer;
```
Comment (liamwhite) [inline src/input_common/helpers/joycon_protocol/common_protocol.h:107]: `std::array<u8, sizeof(Output)> buffer`

Comment (german77) [reply src/input_common/helpers/joycon_protocol/common_protocol.h:107]: ReadSPI can't take an array of fixed size.

**Review by liamwhite — APPROVED** [review]

**Pull request merged**